pkg/operator/operator.go

```diff
@@ -912,28 +912,30 @@ func (o *Operator) Config(ctx context.Context, key string) (*manifests.Config, e
 		}
 	}
 
-	cm, err := o.client.GetConfigmap(ctx, "openshift-config", "etcd-metric-serving-ca")
-	if err != nil {
-		klog.Warningf("Error loading etcd CA certificates for Prometheus. Proceeding with etcd disabled. Error: %v", err)
-		return c, nil
-	}
+	if !o.lastKnowInfrastructureConfig.hostedControlPlane {
```

Comment:
we can add a second check for version 4.14+, as[ commented in the JIRA ticket,](https://issues.redhat.com//browse/OCPBUGS-6677) etcd ServiceMonitor will be removed in 4.14 and newer versions:
```
From version 4.14, CMO will no longer manage etcd ServiceMonitors (will not check for the configmaps/secrets), see https://issues.redhat.com/browse/MON-669
``` 
So that we can put the patch in master branch and backport it to release-4.13 branch, to avoid potential future merge conflicts.